### PR TITLE
Fix Issue 19053 - debug should escape @safe

### PR DESCRIPTION
**Pull request by wilzbach**
- https://issues.dlang.org/show_bug.cgi?id=18407
- https://github.com/dlang/dmd/pull/7882
- https://github.com/dlang/dlang.org/pull/2209

### CODE REVIEW

File: changelog/debug-unsafe.dd [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+When writing debug code, one isn't interested in the type safety, but a pleasant
+debugging experience.
+The type checker already allowed to escape `pure` and `@nogc` within `debug`
+statement.
+With this release, `@system` code can be called from `debug` statements too:
+
+---
+debug unsafeCode();
+
+@system void unsafeCode(){}
```
Comment (Geod24) [inline changelog/debug-unsafe.dd]: I would nest this inside a function for clarity.

**Comment by Geod24** [comment]
Is `-dip1000` checked for in `-debug` ?

**Comment by wilzbach** [comment]
> Is -dip1000 checked for in -debug ?

Well, -dip1000 can mean many things but as long as just `@safe` is concerned safety it should no longer be checked in `debug` too as while the inference will infer a method as unsafe / @system, it should still bypass `checkSafety`. I added an additional test.

File: test/compilable/test16492.d [hunk 2]
```diff
@@ -13,3 +13,26 @@ void test() @nogc pure
...
+void unsafe() @system {}
+void unsafe2()() {
+    int[] arr;
+    auto b = arr.ptr;
+}
+
+void debugSafe2() @safe
+{
+    char[] arr1, arr2;
+    debug unsafe3(arr1, arr2);
```
Comment (Geod24) [inline test/compilable/test16492.d]: I meant something like:
```D
void debugSafe2() @safe
{
    char*ptr;
    char[] arr;
    debug ptr = arr.ptr;
}
```

Comment (wilzbach) [reply test/compilable/test16492.d]: Ah sorry. Yeah that was indeed still forbidden. Fixed.

File: test/compilable/test16492.d [hunk 3]
```diff
@@ -13,3 +13,30 @@ void test() @nogc pure
...
+
+    char*ptr;
```
Comment (Geod24) [inline test/compilable/test16492.d]: Oops I was missing a space. Also, don't make functions needlessly templated.

**Review by RazvanN7 — APPROVED** [review]

**Comment by Geod24** [comment]
@wilzbach : Has the specs been updated ?

**Pull request merged**